cpu/cortexm_common/include/bit.h

```diff
@@ -65,6 +65,20 @@ static inline volatile void *bitband_addr(volatile void *ptr, uintptr_t bit)
         ((((uintptr_t)ptr) & 0xFFFFFul) << 5) + (bit << 2));
 }
 
+/**
+ * @brief Convert memory mapped register address and bit number to bit band alias address
+ *
+ * @param[in] ptr  base address in non bit banded memory-mapped peripheral register
+ * @param[in] bit  bit number within the register
+ *
+ * @return Address of the bit within the bit band memory region
+ */
+static inline volatile void *bitband_hw_addr(volatile void *ptr, uintptr_t bit)
```

Comment:
Is `void *` really the type we want here? The bit band aliases are all word sized, right? So maybe `volatile uint32_t` would be more convenient to use, as you could do

```C
*bitband_hw_addr(foo, 32) = 1;
```

Btw: Do you mind to fix the command above sating that Cortex M3 and higher supports this? This feature was dropped e.g. in Cortex M7 CPUs again.

Reply:
> Is `void *` really the type we want here? The bit band aliases are all word sized, right? So maybe `volatile uint32_t` would be more convenient to use, as you could do
> 
> ```C
> *bitband_hw_addr(foo, 32) = 1;
> ```

That would have saved me a lot of copy & paste.
I thought the most easy route would be to just copy the existing code for SRAM bit-banding and change the name & address.

> Btw: Do you mind to fix the command above sating that Cortex M3 and higher supports this? This feature was dropped e.g. in Cortex M7 CPUs again.

Yea tha comment is wrong.
It's up to the vendor to implement it, e.g on ~~Atmel~~ Microchip MCUs it never available.
I first learned and used this feature on STM32, here we can check for `PERIPH_BB_BASE` in the vendor file.

Reply:
OK. Lets add it like this for consistency.

Reply:
Heh looks like we can save us this whole ordeal: The bit-band address space is `0x2000000` above the base address space for the memory region.
e.g. `0x22000000` for SRAM (`0x20000000`) and `0x42000000` for periphery (`0x40000000`).

Now our `bitband_addr` function will mask all the lower bits of the destination address and add `0x2000000`.

This way we don't even need separate macros for peripherals and sram space, it will all be determined automatically at run-time.

Now if the parmeters of this are not compile time const, this will add a few instructions of overhead compared to what you find everywhere online

```C
#define BITBAND_SRAM_REF   0x20000000
#define BITBAND_SRAM_BASE  0x22000000
#define BITBAND_SRAM(a,b)  (*((volatile uint8_t *) ((BITBAND_SRAM_BASE + ((uintptr_t) (a)-BITBAND_SRAM_REF) << 5 + (b << 2)))))
#define BITBAND_PERI_REF   0x40000000
#define BITBAND_PERI_BASE  0x42000000
#define BITBAND_PERI(a,b)  (*((volatile uint8_t *) ((BITBAND_PERI_BASE + ((uintptr_t) (a)-BITBAND_PERI_REF) << 5 + (b << 2)))))
```

but it means we can reduce this PR to fixing the documentation & ifdefs.